amazon/ion/writer_text.py

```diff
@@ -369,7 +369,7 @@ def _raw_writer_coroutine(depth=0, container_event=None, whence=None, indent=Non
         ion_event, self = (yield transition)
         delegate = self
 
-        if has_written_values and not ion_event.event_type.ends_container:
+        if has_written_values and ((indent and trailing_commas) or not ion_event.event_type.ends_container):
```

Comment:
I may be wrong but I think this will also add a space after the last item in an s-exp: `(foo bar bar )` I'm not sure that's _bad_ but I don't think it's intended.

Reply:
Hm, I did think you were right[1], but I actually think not, because there's a separate set of delimiters defined for use when pretty printing:
```
_CONTAINER_DELIMITER_MAP_PRETTY = {
    IonType.STRUCT: b',',
    IonType.LIST: b',',
    IonType.SEXP: b'', # we use newlines when pretty printing
}
```
which uses an empty string as the delimiter for s-expressions, since they add newlines.

If we **do** want to apply this to **non**-pretty-printing, then we **could** do that, but then your concern expressed here **would** need an additional work-around, or there would indeed be an extra space.

I think the solution, if needed, would be to fork:
```
_CONTAINER_DELIMITER_MAP_NORMAL = {
    IonType.STRUCT: b',',
    IonType.LIST: b',',
    IonType.SEXP: b' ',
}
```

into:
```
_CONTAINER_DELIMITER_MAP_NORMAL = {
    IonType.STRUCT: b',',
    IonType.LIST: b',',
    IonType.SEXP: b' ',
}
_CONTAINER_END_DELIMITER_MAP_NORMAL = {
    IonType.STRUCT: b',',
    IonType.LIST: b',',
    IonType.SEXP: b'',
}
```

and use `_CONTAINER_END_DELIMITER_MAP_NORMAL` for the final trailing delimiter.